pkg/diff/diff.go

```diff
@@ -443,31 +431,33 @@ func normalizeEndpoint(un *unstructured.Unstructured) {
 	var ep corev1.Endpoints
 	err := runtime.DefaultUnstructuredConverter.FromUnstructured(un.Object, &ep)
 	if err != nil {
+		o.log.Error(err, "Failed to convert from unstructured into Endpoints")
 		return
 	}
 	var coreEp core.Endpoints
 	err = v1.Convert_v1_Endpoints_To_core_Endpoints(&ep, &coreEp, nil)
 	if err != nil {
-		log.Warnf("Could not convert from v1 to core endpoint type %s: %v", gvk, err)
+		o.log.Error(err, "Could not convert from v1 to core endpoint type %s", gvk)
 		return
 	}
 
 	endpoints.SortSubsets(coreEp.Subsets)
 
 	err = v1.Convert_core_Endpoints_To_v1_Endpoints(&coreEp, &ep, nil)
 	if err != nil {
-		log.Warnf("Could not convert from core to vi endpoint type %s: %v", gvk, err)
+		o.log.Error(err, "Could not convert from core to vi endpoint type %s", gvk)
 		return
 	}
-	un.Object, err = runtime.DefaultUnstructuredConverter.ToUnstructured(&ep)
```

Comment:
I think this is a bug! We don't want to `nil`-out `un.Object` if there was an error, right?

Reply:
I think you are right. 

@wtam2018 , I think implemented fix is correct, right?

Reply:
Yes, lgtm.   Thanks for the fix.  @ash2k